kayak_widgets/src/text_box.rs

```diff
@@ -0,0 +1,105 @@
+use kayak_ui::core::{
+    render_command::RenderCommand,
+    rsx,
+    styles::{Style, StyleProp, Units},
+    widget, Bound, Color, EventType, MutableBound, OnEvent,
+};
+use std::sync::{Arc, RwLock};
+
+use crate::{Background, Clip, Text};
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct ChangeEvent {
+    pub value: String,
+}
+
+#[derive(Clone)]
+pub struct OnChange(pub Arc<RwLock<dyn FnMut(ChangeEvent) + Send + Sync + 'static>>);
+
+impl OnChange {
+    pub fn new<F: FnMut(ChangeEvent) + Send + Sync + 'static>(f: F) -> OnChange {
+        OnChange(Arc::new(RwLock::new(f)))
+    }
+}
+
+impl PartialEq for OnChange {
+    fn eq(&self, _other: &Self) -> bool {
+        true
+    }
+}
+
+impl std::fmt::Debug for OnChange {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_tuple("OnChange").finish()
+    }
+}
+
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub struct Focus(pub bool);
+
+#[widget(focusable)]
+pub fn TextBox(value: String, on_change: Option<OnChange>) {
+    *styles = Some(Style {
+        render_command: StyleProp::Value(RenderCommand::Layout),
+        ..styles.clone().unwrap_or_default()
+    });
+
+    let background_styles = Style {
+        background_color: StyleProp::Value(Color::new(0.176, 0.196, 0.215, 1.0)),
+        border_radius: StyleProp::Value((5.0, 5.0, 5.0, 5.0)),
+        height: StyleProp::Value(Units::Pixels(26.0)),
+        padding_left: StyleProp::Value(Units::Pixels(5.0)),
+        padding_right: StyleProp::Value(Units::Pixels(5.0)),
+        ..styles.clone().unwrap_or_default()
+    };
+
+    let internal_value = context.create_state("".to_string()).unwrap();
```

Comment:
Shouldn't this be:

```suggestion
    let internal_value = context.create_state(value.clone()).unwrap();
```

to account for the `value` starting as a non-empty String?